src/main/resources/org/apidesign/polyfill/websocket/websocket-polyfill.js

```diff
@@ -1,24 +1,221 @@
+            get: function () {
+                return this.protocols_ ? this.protocols_[0] : '';
+            },
+            enumerable: false,
+            configurable: true
+        });
+
+        Object.defineProperty(WebSocket.prototype, "readyState", {
+            get: function () {
+                return this.state_;
+            },
+            enumerable: false,
+            configurable: true
+        });
+
+        Object.defineProperty(WebSocket.prototype, "binaryType", {
+            value: "arraybuffer",
+            writable: true,
+            enumerable: false,
+            configurable: true
+        });
+
+        //
+        // Listeners
+        //
+
+        Object.defineProperty(WebSocket.prototype, "onopen", {
+            get: function () {
+                return jvm('get-on-listener', 'open');
+            },
+            set: function (listener) {
+                jvm('set-on-listener', 'open', listener);
+            },
+            enumerable: false,
+            configurable: true
+        });
+
+        Object.defineProperty(WebSocket.prototype, "onclose", {
+            get: function () {
+                return jvm('get-on-listener', 'close');
+            },
+            set: function (listener) {
+                jvm('set-on-listener', 'close', listener);
+            },
+            enumerable: false,
+            configurable: true
+        });
+
+        Object.defineProperty(WebSocket.prototype, "onmessage", {
+            get: function () {
+                return jvm('get-on-listener', 'message');
+            },
+            set: function (listener) {
+                jvm('set-on-listener', 'message', listener);
+            },
+            enumerable: false,
+            configurable: true
+        });
+
+        Object.defineProperty(WebSocket.prototype, "onerror", {
+            get: function () {
+                return jvm('get-on-listener', 'error');
+            },
+            set: function (listener) {
+                jvm('set-on-listener', 'error', listener);
+            },
+            enumerable: false,
+            configurable: true
+        });
+
+
+        //
+        // Methods
+        //
+
+        WebSocket.prototype.send = function (data) {
+            if (typeof data.valueOf() === 'string') {
+                jvm('web-socket-send-text', this.connection_, data);
+            } else {
+                jvm('web-socket-send-binary', this.connection_, data);
+            }
+        };
+
+        WebSocket.prototype.close = function (code, reason) {
+            this.state_ = WebSocket.CLOSING;
+            if (code === undefined) {
+                jvm('web-socket-terminate', this.connection_);
+            } else {
+                jvm('web-socket-close', this.connection_, code, reason);
+            }
+        }
+
+        //
+        // EventTarget
+        //
+
+        WebSocket.prototype.addEventListener = function (type, listener) {
```

Comment:
Not using `defineProperty`! Why? Why do we use `defineProperty` once and not in this case?

Reply:
I only used `defineProperty` for fields to make them read-only

Reply:
Why are some fields read only and some are not?

Reply:
In this case it's a function, not a field